src/main/groovy/com/readytalk/gradle/plugins/ci/CiPublishingPlugin.groovy

```diff
@@ -20,8 +20,9 @@ class CiPublishingPlugin implements Plugin<Project>, PluginUtils {
 
   void apply(final Project project) {
     this.project = project
+    project.plugins.apply(BasePlugin)
 
-    withAnyPlugin(['ivy-publish', 'maven-publish']) {
+    withAnyPlugin(['ivy-publish', 'maven-publish', 'com.gradle.plugin-publish']) {
```

Comment:
You'd run into the same problem if you didn't use the helper: how to ensure that a piece of logic is run if any of a set of independent plugins is applied, and ensure that it is only run once.

It's doable with a external counter variable, but I couldn't figure out a way to do that in-line to the function since trying to manipulate the closures too much caused issues with Gradle's internals.